src/Mover.ts

```diff
@@ -796,24 +834,103 @@ export class MoverAPI implements Types.MoverAPI {
         const tabster = this._tabster;
         const focused = tabster.focusedElement.getFocusedElement();
 
-        if (!focused || (await this._isIgnoredInput(focused, keyCode))) {
+        if (focused && (await this._isIgnoredInput(focused, keyCode))) {
             return;
         }
 
-        const ctx = RootAPI.getTabsterContext(tabster, focused, {
-            checkRtl: true,
-        });
+        const ctx =
+            focused &&
+            RootAPI.getTabsterContext(tabster, focused, {
+                checkRtl: true,
+            });
 
-        if (
-            !ctx ||
-            !ctx.mover ||
-            ctx.isExcludedFromMover ||
-            ctx.ignoreKeydown(event)
-        ) {
+        const mover = ctx?.mover;
+
+        if (!mover) {
```

Comment:
4 level deep nesting inside this statement. You got me lost here 😵‍💫, can we break this into more functions or reduce the nesting somehow with some short circuiting?